File: ProblemSolve/53.cpp
/********************
 * Maximum Subarray *
 ********************/

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = -10001;
        int tempSubarray = 0;
        
        //Kadane Algorithm
        for(int i=0; i<nums.size(); ++i){
            tempSubarray += nums[i];
            ans = max(ans, tempSubarray);
            tempSubarray = max(tempSubarray, 0);
        }
        
        return ans;
    }
};

File: ProblemSolve/53.cs
/********************
 * Maximum Subarray *
 ********************/

public class Solution {
    public int MaxSubArray(int[] nums) {
        //-10000 <= nums[i] <= 10000
        int ans = -10001, tempSubarray = 0;
        
        //Kadane Algorithm
        foreach(int num in nums){
            tempSubarray += num;
            ans = Math.Max(ans, tempSubarray);
            tempSubarray = Math.Max(tempSubarray, 0);
        }
        
        return ans;
    }
}